in-toto-spec.md

```diff
@@ -499,18 +498,22 @@ All keys have the format:
 
 ```json
   { "keytype" : "<KEYTYPE>",
+    "scheme" : "<SCHEME>",
     "keyval" : "<KEYVAL>" }
 ```
 
-where KEYTYPE is a string describing the type of the key, and how it is used to
-sign documents. The type determines the interpretation of KEYVAL.
+KEYTYPE is a string denoting a public key signature system, such as RSA or
+ECDSA. SCHEME is a string denoting a corresponding signature scheme. For
+example: "rsassa-pss-sha256" and "ecdsa-sha2-nistp256". KEYVAL is a dictionary
+containing the public portion of the key.
 
-We define two key types at present: 'rsa' and 'ed25519'.
+We define three key types at present: 'rsa', 'ed25519', and 'ecdsa'.
```

Comment:
FYI the in-toto reference implementation does not support the "keytype"-value of "ecdsa". The following values for "keytype" are supported in securesystemslib.
- "rsa"
-  "ecdsa-sha2-nistp256" 
-  "ecdsa-sha2-nistp384"
- "ed25519"
... and for gpg keys 
- "rsa"
- "dsa"
- "eddsa"

As noted in the the commit message of 761aded9968cb380004a86c44eb4796c061bca9c I think we should reconsider the "key type" vs. "signature scheme" nomenclature for ecdsa keys. The way we do it here in the spec, e.g. keytype=ecdsa and scheme=ecdsa-sha2-nistp256 makes more sense to me. Should we create a ticket in securesystemslib?

Reply:
I agree, I think this needs to be resolved, and we should separate out keytype and scheme. I see a possible source of confusion in the TUF spec as well:

```
KEYTYPE is a string denoting a public key signature system, such as RSA or ECDSA.
...
We define three keytypes below: 'rsa', 'ed25519', and 'ecdsa', but adopters can define and use any particular keytype, signing scheme, and cryptographic library.
```

...but it's clearer afterwards, of course. I'm hesitant to expand the keytype here in the meantime if we're looking to fix it in securesystemslib in the near future, as we're looking to tag the spec at 1.0 soon, and I'm not sure if this is a change that makes things a problem. What do you think?

Reply:
> I'm hesitant to expand the keytype here 

Agreed, this seems like something that should be fixed in securesystemslib. Do you mind submitting an issue there? :)

Reply:
https://github.com/secure-systems-lab/securesystemslib/issues/239 :)